### ISSUE TITLE
Any help on making it run

### ISSUE THREAD

**Issue by machinexa2** [ultrafunkamsterdam/undetected-chromedriver#653]
OS: Debian Sid
```
root@kali-linux:~/Temp/Cloudflare# python3 --version
Python 3.10.4
root@kali-linux:~/Temp/Cloudflare# pip show undetected-chromedriver
Name: undetected-chromedriver
Version: 3.1.5.post4
Summary: ('Selenium.webdriver.Chrome replacement with compatiblity for Brave, and other Chromium based browsers.', 'Not triggered by CloudFlare/Imperva/hCaptcha and such.', 'NOTE: results may vary due to many factors. No guarantees are given, except for ongoing efforts in understanding detection algorithms.')
Home-page: https://github.com/ultrafunkamsterdam/undetected-chromedriver
Author: UltrafunkAmsterdam
Author-email: info@blackhat-security.nl
License: GPL-3.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: requests, selenium, websockets
Required-by:
```
```
root@kali-linux:~/Temp/Cloudflare# chromium --version
Chromium 101.0.4951.64 built on Debian bookworm/sid, running on Debian kali-rolling
root@kali-linux:~/Temp/Cloudflare# chromedriver --version
ChromeDriver 101.0.4951.41 (93c720db8323b3ec10d056025ab95c23a31997c9-refs/branch-heads/4951@{#904})
```
```
Traceback (most recent call last):
  File "/root/Temp/Cloudflare/Cloudflare.py", line 10, in <module>
    driver = uc.Chrome(options=options)
  File "/usr/local/lib/python3.10/dist-packages/undetected_chromedriver/__init__.py", line 401, in __init__
    super(Chrome, self).__init__(
  File "/usr/local/lib/python3.10/dist-packages/selenium/webdriver/chrome/webdriver.py", line 70, in __init__
    super(WebDriver, self).__init__(DesiredCapabilities.CHROME['browserName'], "goog",
  File "/usr/local/lib/python3.10/dist-packages/selenium/webdriver/chromium/webdriver.py", line 92, in __init__
    RemoteWebDriver.__init__(
  File "/usr/local/lib/python3.10/dist-packages/selenium/webdriver/remote/webdriver.py", line 275, in __init__
    self.start_session(capabilities, browser_profile)
  File "/usr/local/lib/python3.10/dist-packages/undetected_chromedriver/__init__.py", line 589, in start_session
    super(selenium.webdriver.chrome.webdriver.WebDriver, self).start_session(
  File "/usr/local/lib/python3.10/dist-packages/selenium/webdriver/remote/webdriver.py", line 365, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
  File "/usr/local/lib/python3.10/dist-packages/selenium/webdriver/remote/webdriver.py", line 430, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python3.10/dist-packages/selenium/webdriver/remote/errorhandler.py", line 247, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: unknown error: cannot connect to chrome at 127.0.0.1:33145
from chrome not reachable
Stacktrace:
#0 0x561013f19f33 <unknown>
#1 0x561013c63faf <unknown>
#2 0x561013c52209 <unknown>
#3 0x561013c8aa79 <unknown>
#4 0x561013c82a06 <unknown>
#5 0x561013cbdd3a <unknown>
#6 0x561013cb7e63 <unknown>
#7 0x561013c8d82a <unknown>
#8 0x561013c8e985 <unknown>
#9 0x561013f5e4cd <unknown>
#10 0x561013f625ec <unknown>
#11 0x561013f4871e <unknown>
#12 0x561013f63238 <unknown>
#13 0x561013f3d870 <unknown>
#14 0x561013f7f608 <unknown>
#15 0x561013f7f788 <unknown>
#16 0x561013f99f1d <unknown>
#17 0x7f72033cdd80 <unknown>

```
Chromedriver is already in path. Any idea why it fails?

**Comment by machinexa2** [comment]
```python
from selenium import webdriver
from selenium_stealth import stealth

options = webdriver.ChromeOptions()
options.add_argument("start-maximized")
options.add_argument("no-sandbox")
#options.add_argument("--headless")
options.add_experimental_option("excludeSwitches", ["enable-automation"])
options.add_experimental_option('useAutomationExtension', False)
driver = webdriver.Chrome(options=options, executable_path=r"/root/MachineYadav/My-Tools/MyBinaries/chromedriver")

stealth(driver,
        languages=["en-US", "en"],
        vendor="Google Inc.",
        platform="Win32",
        webgl_vendor="Intel Inc.",
        renderer="Intel Iris OpenGL Engine",
        fix_hairline=True,
)

driver.get("https://bot.sannysoft.com/")
```
works but eventually gets detected by Cloudflare. I think this module will not get detected

**Comment by sebdelsol** [comment]
It seems that you use a specific chromedriver which defeats undetected-chromedriver purpose since you'll end up with a driver having always the same signature. Anyway in this case so you've to be sure it's the same main version as your browser.

**Comment by machinexa2** [comment]
How to use your custom version Chromedriver?

**Comment by machinexa2** [comment]
Also note that experimental options fails for certain reason so I have disabled them as well

**Comment by machinexa2** [comment]
How to solve the problem?

**Comment by sebdelsol** [comment]
* Only `'prefs'` experimental options works
* The experimental options you use are not needed for not being detected
```python3
import undetected_chromedriver as uc

if __name__ == "__main__":
    options = uc.ChromeOptions()
    options.add_argument("start-maximized")
    driver = uc.Chrome(options=options)
    driver.get("https://bot.sannysoft.com/")
```

**Comment by machinexa2** [comment]
hmm
